Datadog/IntegrationUnitTests/Public/CoreTelemetryIntegrationTests.swift

```diff
@@ -204,4 +204,51 @@ class CoreTelemetryIntegrationTests: XCTestCase {
         XCTAssertNotNil(methodCalledMetric.view?.id)
         XCTAssertNotNil(methodCalledMetric.action?.id)
     }
+    
+    func testGivenRUMEnabled_effectiveSampleRateIsComposed() throws {
+        // Given
+        var config = RUM.Configuration(applicationID: .mockAny())
+        config.telemetrySampleRate = 90
+        RUM.enable(with: config, in: core)
+
+        // When
+        core.telemetry.debug("Debug Telemetry")
+        core.telemetry.error("Error Telemetry")
+        core.telemetry.metric(name: "Metric Name", attributes: [:], sampleRate: 99)
+        core.telemetry.send(telemetry: .usage(.init(event: .setUser, sampleRate: 100)))
+        core.telemetry.stopMethodCalled(
+            core.telemetry.startMethodCalled(operationName: .mockRandom(), callerClass: .mockRandom(), headSampleRate: 80),
+            tailSampleRate: 99
+        )
+
+        // Then
+        let debugEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryDebugEvent.self)
+        let errorEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryErrorEvent.self)
+        let usageEvents = core.waitAndReturnEvents(ofFeature: RUMFeature.name, ofType: TelemetryUsageEvent.self)
+        
+        if let debug = debugEvents.first,
+           debug.telemetry.message == "Debug Telemetry" {
+         
+            XCTAssertEqual(debug.effectiveSampleRate, 90)
+        }
+        
+        if let error = errorEvents.first {
+            XCTAssertEqual(error.telemetry.message, "Error Telemetry")
+            XCTAssertEqual(error.effectiveSampleRate, 90)
+        }
```

Comment:
suggestion/ As I understand, we use if let here because the events may not be sent due to the sample rate values we are testing, which makes sense 👍.

However, this approach could become unreliable if an issue is introduced, but no events are sampled, meaning assertions won’t catch potential regressions. In such cases, CI could pass even though the test might fail at another time due to missed samples.

To mitigate this, we could increase the sample rate and send more events, reducing the likelihood of having zero samples in the resulting arrays. By sending multiple telemetry events, the chances of not having a single sample would become negligible, and we could safely assert that the events are sent.

Here’s an example of how this can be done:
```swift
// When
for idx in (0..<100) {
    core.telemetry.debug("Debug Telemetry \(idx)")
    core.telemetry.error("Error Telemetry \(idx)")
    core.telemetry.metric(name: "Metric Name", attributes: [:], sampleRate: 99)
    core.telemetry.send(telemetry: .usage(.init(event: .setUser, sampleRate: 100)))
    core.telemetry.stopMethodCalled(
        core.telemetry.startMethodCalled(operationName: .mockRandom(), callerClass: .mockRandom(), headSampleRate: 80),
        tailSampleRate: 99
    )
}
```

With this approach, the chances of not having a single sample in the resulting arrays are extremely low, allowing us to assert:
```swift
// Then
XCTAssertGreaterThan(debugEvents.count, 0)
XCTAssertGreaterThan(errorEvents.count, 0)
XCTAssertGreaterThan(usageEvents.count, 0)

// loop through each and assert on `effectiveSampleRate`
```

**Rationale:**
Given an effectiveSampleRate of 80%, the probability of having no sample in 100 consecutive attempts is `(0.2)^100`, which is effectively zero. Therefore, we can confidently assert the presence of events in the arrays.

Reply:
Good point here 🙏  It could hide some regressions. Changed based on the suggestion!